src/sage/rings/function_field/drinfeld_modules/drinfeld_module.py

```diff
@@ -883,6 +891,209 @@ def action(self):
+             ((1, 2), (7, 4, 1)),
+             ((1, 2), (13, 3, 1)),
+             ((1, 2), (19, 2, 1)),
+             ((1, 2), (25, 1, 1)),
+             ((1, 2), (31, 0, 1)),
+             ((1, 2), (8, 9, 2)),
+             ((1, 2), (20, 7, 2)),
+             ((1, 2), (9, 14, 3)),
+             ((1, 2), (15, 13, 3)),
+             ((1, 2), (27, 11, 3)),
+             ((1, 2), (10, 19, 4)),
+             ((1, 2), (22, 17, 4)),
+             ((1, 2), (11, 24, 5)),
+             ((1, 2), (17, 23, 5)),
+             ((1, 2), (23, 22, 5)),
+             ((1, 2), (29, 21, 5)),
+             ((1, 2), (0, 31, 6)),
+             ((1, 2), (12, 29, 6)),
+             ((1, 2), (31, 31, 7))]
+
+        Specify a list of indices to restrict which coefficients appear
+        in the parameters::
+
+            sage: A = GF(3)['T']
+            sage: K.<T> = Frac(A)
+            sage: phi = DrinfeldModule(A, [T, T, 2, T, 2*T, T^3, T^4 + T^2 + 1])
+            sage: phi.basic_j_invariant_parameters([1, 5])
+            [((1, 5), (273, 91, 31)),
+             ((1, 5), (297, 163, 55)),
+             ((1, 5), (295, 157, 53)),
+             ((1, 5), (265, 67, 23)),
+             ((1, 5), (357, 343, 115)),
+             ...
+             ((1, 5), (146, 74, 25))]
+
+        Use ``nonzero=True`` to speed up the computations for Drinfeld
+        modules having multiple zero coefficients::
+
+            sage: A = GF(5)['T']
+            sage: K.<T> = Frac(A)
+            sage: phi = DrinfeldModule(A, [T, 0, T+1, 0, 1, 0, T])
+            sage: phi.basic_j_invariant_parameters(nonzero=True)
+            [((2, 4), (260, 641, 26)),
+             ((2, 4), (157, 19, 1)),
+             ((2, 4), (27, 24, 1)),
+             ((2, 4), (188, 143, 6)),
+             ((2, 4), (401, 260, 11)),
+             ...
+             ((2, 4), (288, 39, 2))]
+
+        TESTS::
+
+            sage: A = GF(5)['T']
+            sage: K.<T> = Frac(A)
+            sage: phi = DrinfeldModule(A, [T, 0, T+1, T^2 + 1])
+            sage: phi.basic_j_invariant_parameters([1, 'x'])
+            Traceback (most recent call last):
+            ...
+            TypeError: coefficients indices must be integers
+            sage: phi.basic_j_invariant_parameters([1, 10])
+            Traceback (most recent call last):
+            ...
+            ValueError: indices must be > 0 and < 3
+            sage: phi.basic_j_invariant_parameters([1, 1])
+            Traceback (most recent call last):
+            ...
+            ValueError: indices must be distinct and sorted
+            sage: phi.basic_j_invariant_parameters([2, 1])
+            Traceback (most recent call last):
+            ...
+            ValueError: indices must be distinct and sorted
+            sage: phi.basic_j_invariant_parameters('x')
+            Traceback (most recent call last):
+            ...
+            TypeError: indices must be None, a tuple or a list
+        """
+        r = self._gen.degree()
```

Comment:
I suggested this in my first review. I really think

```suggestion
        r = self.rank()
```

is a huge bit readability gain, especially for users that are not familiar with the codebase. The cost of calling `rank` is negligible (and we are not yet at this level of optimization). However, `self.rank()` reads *way* better than `self._gen.degree()`. As developers, it is our duty to create code that is both for machines (computational efficiency) and for human beings (reading ease and comfort). As [Guido points out](https://pep8.org/#a-foolish-consistency-is-the-hobgoblin-of-little-minds):

> code is read much more often than it is written

Last but not least, this would be more consistent with the codebase.

This being said, I would agree that this is a minor detail, and I won't insist afterwards.

Reply:
Yes that's a minor detail and I think that it is not a very huge readability gain. I would prefer to keep accessing the attribute directly.